### PR TITLE
Resolve #1052

### LINKED ISSUE #1052: vtm writer not expanding user path
When saving a VTM (`pyvista.MultiBlock`) data object to a `vtm` archive the user path `~` is not expanded and you end up with a new directory `~/*` inn your working directory. This is inconsistent with other `save` methods on objects.

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Resolve #1052

 

### Details

As far as I see, all we have to fix is only `DataObject` (https://github.com/pyvista/pyvista/issues/1052#issuecomment-775566180).

### CODE REVIEW

**Review by akaszynski — CHANGES_REQUESTED** [review]
We're writing to the user path during testing.  Unit tests are all done in the isolated `tmpdir` environment to avoid any conflicts with existing files.  If (for whatever reason) the user had a `~/tmp.vtk` file, we'd overwrite it.

Best approach I think is to simply not test this functionality within our unit tests.  I don't like it, but it's better than writing temp files to user paths.

**Comment by tkoyama010** [comment]
@akaszynski 

> We're writing to the user path during testing. Unit tests are all done in the isolated tmpdir environment to avoid any conflicts with existing files. If (for whatever reason) the user had a `~/tmp.vtk` file, we'd overwrite it.
> Best approach I think is to simply not test this functionality within our unit tests. I don't like it, but it's better than writing temp files to user paths.

I don't like either. However, it is not good to write in the user path as you say :-( I will delete the test. Thank you for reivew.

File: tests/test_grid.py [hunk 1]
```diff
@@ -829,3 +829,17 @@ def test_hide_cells(ind, struct_grid):
...
+    hexbeam.save(filename)
+
+    grid = pyvista.UnstructuredGrid(filename)
+    assert grid.cells.shape == hexbeam.cells.shape
+    assert grid.points.shape == hexbeam.points.shape
+
+    grid = pyvista.read(filename)
+    assert grid.cells.shape == hexbeam.cells.shape
+    assert grid.points.shape == hexbeam.points.shape
+    assert isinstance(grid, pyvista.UnstructuredGrid)
```
Comment (tkoyama010) [inline tests/test_grid.py:845]:
```suggestion
```

File: tests/test_polydata.py [hunk 2]
```diff
@@ -729,3 +729,17 @@ def test_flip_normals(sphere):
...
+    sphere.save(filename)
+
+    mesh = pyvista.PolyData(filename)
+    assert mesh.faces.shape == sphere.faces.shape
+    assert mesh.points.shape == sphere.points.shape
+
+    mesh = pyvista.read(filename)
+    assert isinstance(mesh, pyvista.PolyData)
+    assert mesh.faces.shape == sphere.faces.shape
+    assert mesh.points.shape == sphere.points.shape
```
Comment (tkoyama010) [inline tests/test_polydata.py:745]:
```suggestion
```

**Review by akaszynski — APPROVED** [review]
Thanks for fixing that.  At least it's hit by code coverage.

**Pull request merged**